### ISSUE TITLE
Timer.delete(null) crash on Android

### ISSUE THREAD

**Issue by eksperts** [fusetools/fuselibs-public#382]
Fuse 1.2.0 build 13961 (bundled libs), macOS. Likely other Fuse versions affected too.

[Reported on community](https://www.fusetools.com/community/forums/bug_reports/data_loaded_the_application_crash_after_updating_f) with a larger reproduction.

Only a problem on Android, other targets appear to be fine (remains to be tested).

Minimal repro:
```
<App>
    <JavaScript>
    var Timer = require('FuseJS/Timer');
    setTimeout(function() {
        Timer.delete(null);
    }, 3000);
    </JavaScript>
</App>
```
Causes a fatal crash in Android app. Throws something like this in Android Studio:
<img width="1423" alt="screen shot 2017-08-17 at 13 30 25" src="https://user-images.githubusercontent.com/7188878/29408292-4e5de836-8350-11e7-85f5-02e63a1ed5ae.png">

**Comment by fusebuild** [comment]
The forum thread [Data loaded, the application crash after updating Fuse 1.2.1](https://www.fusetools.com/community/forums/bug_reports/data_loaded_the_application_crash_after_updating_f) posted by Jae Woong Kang was linked to this issue.

**Comment by fusebuild** [comment]
The forum thread [socket.io!](https://www.fusetools.com/community/forums/show_and_tell/socketio) posted by Bjørn-Olav Strand was linked to this issue.

**Comment by fusebuild** [comment]
The forum thread [how to use socket io with fusetools?](https://www.fusetools.com/community/forums/howto_discussions/how_to_use_socket_io_with_fusetools) posted by Pasindu Dineth Peiris was linked to this issue.

**Comment by eksperts** [comment]
This issue also causes socket.io library to fail hard, since their implementation apparently also ends up trying to delete a NULL timer someplace. That is 3rd party code, and not easily fixable.

I believe we should add some safeguarding here, so that this doesn't cause a fatal crash on Android.

**Comment by igorls** [comment]
I've managed to fix the socket.io lib to solve the issues on android, actually the issue lies on the engine.io dependency, exactly on this file (https://github.com/socketio/engine.io/blob/master/lib/socket.js).

There are several clearTimeout call that are not protected from receiving null

Here's my fixed version based on socket.io 2.0.3 client library for anyone interested in testing, it's not uglyfied for debugging.
I've tested superficially on desktop preview and android
https://gist.github.com/igorls/5009b33230bb635590bebf364f14cc45

**Comment by PasinduDineth** [comment]
@igorls I have tested your library with macOS+samsung galaxy s5. **App is still crashing** after few seconds. since i am one of the person who reported this issue, i would like to help you to solve this issue. So is there any information i can provide you? can you add the link of your tested fuse app code and server code? so i can run it in my mobile and mac to clarify the issue. Thank you!

**Comment by igorls** [comment]
@PasinduDineth can you do a debug on the V8 engine? We have to know what is going on when you app crashes. Have you tested on preview or export?

**Comment by igorls** [comment]
@PasinduDineth I've made some other fixes, please try again with the updated version https://gist.github.com/igorls/5009b33230bb635590bebf364f14cc45

**Comment by PasinduDineth** [comment]
@igorls this is the error i am getting.
```
09-15 16:10:54.091 19833 19886 F libc    : Fatal signal 11 (SIGSEGV), code 1, fault addr 0x7 in tid 19886 (Thread-20339)
09-15 16:10:54.161 26499 26499 F DEBUG   : pid: 19833, tid: 19886, name: Thread-20339  >>> com.apps.appname <<<
09-15 16:10:55.081   813 25687 D InputDispatcher: Focus left window: 19833
09-15 16:10:55.191   813  1259 W InputMethodManagerService: Got RemoteException sending setActive(false) notification to pid 19833 uid 10431
09-15 16:10:55.201   353   353 I Zygote  : Process 19833 exited due to signal (11)
09-15 16:10:55.201   813  1736 I ActivityManager: Process com.apps.appname (pid 19833)(adj 0) has died(283,326)
09-15 16:10:55.201   813  1736 D ActivityManager: cleanUpApplicationRecord -- 19833
Process com.apps.appname terminated.
```

**Comment by ruk91** [comment]
@igorls this is the error I am getting when I use V8 debugger.
```
09-15 17:03:23.231   813  1236 I InputDispatcher: Delivering touch to (10552): action: 0x0, toolType: 1
09-15 17:03:23.231 10552 10552 D ViewRootImpl: ViewPostImeInputStage processPointer 0
09-15 17:03:23.341   813  1236 I InputDispatcher: Delivering touch to (10552): action: 0x1, toolType: 1
09-15 17:03:23.351 10552 10552 D ViewRootImpl: ViewPostImeInputStage processPointer 1
09-15 17:03:23.411 10552 10623 F libc    : Fatal signal 11 (SIGSEGV), code 1, fault addr 0x7 in tid 10623 (Thread-21731)
09-15 17:03:23.471 26499 26499 F DEBUG   : pid: 10552, tid: 10623, name: Thread-21731  >>> com.apps.appname <<<
09-15 17:03:24.311   813 11121 D InputDispatcher: Focus left window: 10552
09-15 17:03:24.361   813  1757 W InputMethodManagerService: Got RemoteException sending setActive(false) notification to pid 10552 uid 10435
09-15 17:03:24.401   353   353 I Zygote  : Process 10552 exited due to signal (11)
09-15 17:03:24.431   292   292 E lowmemorykiller: Error opening /proc/10552/oom_score_adj; errno=2
09-15 17:03:24.481   813  1751 I ActivityManager: Process com.apps.appname (pid 10552)(adj 1) has died(275,324)
09-15 17:03:24.481   813  1751 D ActivityManager: cleanUpApplicationRecord -- 10552
Process com.apps.appname terminated.

```

**Comment by igorls** [comment]
@ruk91 that was with the latest version? Can you give more info on which fuse version, build target, OS?

**Comment by ruk91** [comment]
Yes, The versions I am using are VS Code 1.16.1 and Fuse 1.2.1. And I was tested with Samsung Galaxy S5 and Nexus 5X in windows platform.

**Comment by igorls** [comment]
Are you able to pinpoint the line on which the js module is crashing?

**Comment by ruk91** [comment]
@igorls  I tried, but I was unable to pinpoint exact line which corresponds for the crash. Is there any method for the pinpoint corresponding line?

**Comment by igorls** [comment]
@Tapped JavaScriptCore will be included in the next release?

@ruk91 I can only suggest you carefully debugging your app, setting the debug tag inside socket.io js, try to notice on which line the error is thrown

**Comment by kusma** [comment]
@igorls: We don't know yet, but it being included in *some* upcoming release seems reasonably likely. If it will stay an opt-in or become the default at some point is another question.
